src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -303,9 +303,6 @@ internal int SerializeLogRecord(LogRecord logRecord)
                     case double vd:
                     case string vs:
                         break;
-                    default:
```

Comment:
I think we still need a `default` case. We should probably take the `.ToString()` value here.

Reply:
After thinking some more about this, I believe that we should move this validation check to the GenevaExporterOptions class in the setter of `PrepopulatedFields`. We can then get rid of the switch case construct here in the exporter classes.

Reply:
What's the design here? E.g. what would happen if the user triggered this code path?

Reply:
Here's the design for one of the possible solutions:

### Problem
We want to validate the data types for `PrepopulatedFields`. As of today, if someone passes in a value with an unsupported data type, we serialize that into a very unhelpful [string](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Exporter.Geneva/GenevaExporterOptions.cs#L98).

### Solution
The best thing to do here would be to validate the data upfront when the user actually provides the `PrepopulatedFields` option. The best place to do that would be the [setter ](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Exporter.Geneva/GenevaExporterOptions.cs#L87) of the `PrepopulatedFields`. We have a set of types that we support as a valid data type for the values. These data types are mentioned as various cases in the `switch`. We need a `default` case which will be hit for any unsupported type. For the `default` case, we throw an `ArgumentException` with the message: `$"Type: {value.GetType()} is not supported. Only bool, byte, sbyte, short, ushort, int, uint, long, ulong, float, double are the supported types for PrepopulatedFields values";` .

We can then get rid of the switch case checks in the individual exporters as we would not get any unsupported data type.

Reply:
@utpilla Agree with the design/solution, would you mind create another issue, then we can track change from there?

Reply:
I don't think we need to create another issue.

@reyang I have updated the existing issue #395 with this [comment](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/514#discussion_r928041421) detailing the problem description and one of the possible solutions. Could you please check if that looks good? Thanks!

Reply:
> Only bool, byte, sbyte, short, ushort, int, uint, long, ulong, float, double are the supported types

and string